### DISCUSSION TITLE
System Message not applied when generating with data

### DISCUSSION THREAD

**Discussion by olivervarosanec** [microsoft/sample-app-aoai-chatGPT#310] [Q&A]
Hello Community,
I've been working with the AZURE_OPENAI_SYSTEM_MESSAGE and have observed its application in conversation_without_data but not in conversation_with_data. 
This makes it challenging to instruct the Chatbot on how to structure or format the retrieved data. 

Is this intentional, or are there technical factors that prevent the system message from being submitted when requesting with data? Additionally, is there a workaround or alternative method that I might be overlooking to navigate through this scenario?
I appreciate any insights or suggestions you may have.

Thank you!

**Comment by sydneybeckcook** [comment]
Hi, I am also wondering what the answer to this is. Was this intentional to not include the system message when using a datasource? If possible, I would like to include both.
